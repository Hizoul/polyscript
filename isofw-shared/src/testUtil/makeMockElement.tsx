import * as React from "react"

const makeMockElement = (name: string, showChildren: boolean = false) => {
  return (props: any) => (
    <div className={name}>
      {JSON.stringify({...props, children: undefined}, undefined, 2)}
      {props.children}
    </div>
  )
  }

export default makeMockElement
