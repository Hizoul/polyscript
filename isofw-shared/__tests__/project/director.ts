import useDirectorComponent from "isofw-shared/src/components/project/directorSheet"
import directorTest from "isofw-shared/src/tests/project/director"
import makeMockElement from "isofw-shared/src/testUtil/makeMockElement"

directorTest(makeMockElement("DirectorEle", (props) => useDirectorComponent(props.id, props.reset)))
