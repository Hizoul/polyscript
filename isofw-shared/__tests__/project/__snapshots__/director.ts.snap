// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` direct a test   can increase number :  info pushed by Server  1`] = `
<div
  className="DirectorEle"
>
  {
  "user": {
    "_id": "u1",
    "email": "u1"
  }
}
</div>
`;

exports[` direct a test   can increase number :  project Edited by client 1`] = `
Object {
  "createState": Object {},
  "currentlyEditing": "b19999999999999999999999",
  "fetching": Object {},
  "getFromServer": [Function],
  "getState": Object {
    "projects": Object {
      "b19999999999999999999999": Object {
        "_id": "b19999999999999999999999",
        "isActive": true,
        "name": "pn1",
        "shot": 43,
      },
    },
    "users": Object {
      "u1": Object {
        "_id": "u1",
        "email": "u1",
      },
    },
  },
  "lastFetch": Object {
    "b19999999999999999999999": -2077750800000,
  },
  "removeState": Object {},
  "updateState": Object {
    "edit.projectModelb19999999999999999999999": Object {
      "_id": "b19999999999999999999999",
      "isActive": true,
      "name": "pn1",
      "shot": 43,
    },
  },
}
`;

exports[` direct a test   can increase number :  project pushed by server  1`] = `
Object {
  "createState": Object {},
  "currentlyEditing": "",
  "fetching": Object {},
  "getFromServer": [Function],
  "getState": Object {
    "projects": Object {
      "b19999999999999999999999": Object {
        "_id": "b19999999999999999999999",
        "isActive": true,
        "name": "pn1",
        "shot": 42,
      },
    },
    "users": Object {
      "u1": Object {
        "_id": "u1",
        "email": "u1",
      },
    },
  },
  "lastFetch": Object {
    "b19999999999999999999999": -2077750800000,
  },
  "removeState": Object {},
  "updateState": Object {},
}
`;

exports[` direct a test   can increase number : After manual edits 1`] = `
<div
  className="DirectorEle"
>
  {
  "state": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 43,
    "isActive": true
  },
  "loading": false,
  "original": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 43,
    "isActive": true
  },
  "user": {
    "_id": "u1",
    "email": "u1"
  }
}
</div>
`;

exports[` direct a test   can increase number : Before manual edits 1`] = `
<div
  className="DirectorEle"
>
  {
  "user": {
    "_id": "u1",
    "email": "u1"
  }
}
</div>
`;
